src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/repository.py

```diff
@@ -42,46 +43,60 @@ def _bearer_auth_str(token):
     return 'Bearer ' + token
 
 
-def _headers(username, password):
+def _headers(username, password, is_manifest=False):
     if username is None:
         auth = _bearer_auth_str(password)
     else:
         auth = _basic_auth_str(username, password)
 
+    if is_manifest:
+        return {'Authorization': auth,
+                'Accept': DOCKER_MANIFEST_V2}
     return {'Authorization': auth}
 
 
-def _delete_data_from_registry(login_server,
-                               path,
-                               username,
-                               password,
-                               resultIndex=None,  # pylint: disable=unused-argument
-                               retry_times=3,  # pylint: disable=unused-argument
-                               retry_interval=5):  # pylint: disable=unused-argument
-    registryEndpoint = 'https://' + login_server
+def _params(count=20):
+    return {'n': count}
 
-    response = requests.delete(
-        registryEndpoint + path,
-        headers=_headers(username, password)
-    )
 
-    if response.status_code == 200 or response.status_code == 202:
-        return
-    elif response.status_code == 401:
-        raise Unauthorized(response.text)
-    elif response.status_code == 404:
-        raise NotFound(response.text)
-    else:
-        raise CLIError(response.text)
+def _delete_data_from_registry(login_server, path, username, password, **kwargs):
+    retry_times = kwargs['retry_times'] if 'retry_times' in kwargs else 3
+    retry_interval = kwargs['retry_interval'] if 'retry_interval' in kwargs else 5
+
+    for i in range(0, retry_times):
+        try:
+            errorMessage = None
+            response = requests.delete(
+                'https://{}/{}'.format(login_server, path),
+                headers=_headers(username, password)
+            )
+
+            if response.status_code == 200 or response.status_code == 202:
```

Comment:
Is this condition too rigorous? Do the other status code in 2XX represents a failure?